lib/flipper/api/v1/actions/groups_gate.rb

```diff
@@ -27,7 +27,23 @@ def delete
           private
 
           def ensure_valid_params
-            json_error_response(:group_not_registered) unless Flipper.group_exists?(group_name)
+            if group_name.nil? || group_name.empty?
+              json_error_response(:name_invalid)
+            end
+
+            return if allow_unregistered_groups?
+            return if Flipper.group_exists?(group_name)
+
+            json_error_response(:group_not_registered)
+          end
+
+          def allow_unregistered_groups?
+            allow_unregistered_groups = params['allow_unregistered_groups']
```

Comment:
@AlexWheeler if you have any time to look this over, I'd be curious what your thoughts are. It will be a few weeks at least before I do another release, so no rush. I went with a specific param instead of `strict`. `strict` would affect everything whereas specific params are more isolated. Specific felt better as a way to make it possible to opt in to specific functionality instead of everything (if there are other things down the road).